pandas/src/period.pyx

```diff
@@ -627,6 +628,11 @@ cdef ndarray[int64_t] localize_dt64arr_to_period(ndarray[int64_t] stamps,
 _DIFFERENT_FREQ = "Input has different freq={1} from Period(freq={0})"
 _DIFFERENT_FREQ_INDEX = "Input has different freq={1} from PeriodIndex(freq={0})"
 
+
+class IncompatibleFrequency(ValueError):
```

Comment:
if its helpful you can import this into ``pandas.tseries.common``. can do that in the future.